paddle/framework/tensor.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+#include <typeinfo>
+#include "paddle/framework/ddim.h"
+#include "paddle/framework/enforce.h"
+#include "paddle/platform/assert.h"
+#include "paddle/platform/place.h"
+
+namespace paddle {
+namespace framework {
+
+class Tensor {
+  using paddle::platform::Place;
+  using paddle::platform::get_place;
+
+ public:
+  explicit Tensor(DDim dims) : dims_(dims), place_(get_place()) {}
+  explicit Tensor(DDim dims, Place place) : dims_(dims), place_(place) {}
+
+  Tensor& operator=(const Tensor& src) = delete;
+
+  template <typename T>
+  const T* data() const {
+    PADDLE_ENFORCE(holder_ != nullptr);
+    PADDLE_ENFORCE(holder_->Place() == place_);
+    PADDLE_ENFORCE(holder_->Size() >= product(dims_) * sizeof(T));
+    return static_cast<const T*>(holder->Ptr());
+  }
+
+  template <typename T>
+  bool NeedReset() const {
```

Comment:
I think @hedaoyuan 's comment means these functions can be inlined by using `inline` keyword.